network/api.go

```diff
@@ -134,6 +134,18 @@ type LeaveRequest struct {
 	Options    map[string]interface{}
 }
 
+// DiscoverNew is a notification for a new discovery event, Example:a new node joining a cluster
+type DiscoverNewRequest struct {
```

Comment:
We should just make this `DiscoverRequest` as the same struct can be used for both New and Delete